tests/python/unittest/test_summary.py

```diff
@@ -0,0 +1,344 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import shutil
+from mxnet.contrib.summary import SummaryWriter
+from mxnet.contrib.summary.utils import _make_metadata_tsv, make_image_grid, _make_sprite_image
+from mxnet.contrib.summary.utils import _add_embedding_config, _save_embedding_tsv
+from mxnet.test_utils import *
+from common import with_seed
+
+# DO NOT CHANGE THESE NAMES AS THEY FOLLOW THE DEFINITIONS IN TENSORBOARD
+_LOGDIR = './logs_for_tensorboard'
```

Comment:
Could you use a tempdir? https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/common.py#L231

Reply:
one problem with using tempdir is that for troubleshooting, no one will know where to look for logs without looking in the code.

Reply:
Good point. How about we introduce a temp-dir in the root of mxnet that is covered by gitignore? At the moment, running tests will result in a lot of files being created.

Reply:
Yes, can we put it in a local temp anyways? E.g. ./TMP/model1.json

Reply:
actually I’m not crazy about a “temp sir”, and I’ll tell you why — try not to look at this from a standpoint of mxnet being an app, because it isn’t.  It’s a library. It runs in whatever directory is the current directory for which it is invoked. you have no idea what that directory may be, and it’s quite possible that whoever is using the library doesn’t have any idea about the logging mechanism and expecting to create a directory TMP on a unix filesystem is a stretch in presumption.  there could be an environment var override, possibly, but it’s easiest for everyone to find it to just dump it in cwd. profiler for example goes to current dir by default.
*.log can be added to .gitignore

Reply:
Very good point, Chris! This sounds reasonable and seems to be a good solution.